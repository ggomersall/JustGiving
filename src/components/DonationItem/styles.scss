.card.donation {
  min-height: 150px;
}

.item-donor {
  display: flex;
  align-items: center;
  margin-bottom: 15px;
}

.item-donor__image {
  margin-right: 20px;
  img {
    border-radius: 50%;
    height: 50px;
  }
}