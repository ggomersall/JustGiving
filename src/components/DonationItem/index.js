import React from 'react';
import PropTypes from 'prop-types';

import './styles.scss';

const DonationItem = ({ data }) => (
  <div className="card donation">
    <div className="item__content">
      <div className="item-donor">
        <div className="item-donor__image">
          <img src={data.imageUrl} />
        </div>
        <div className="item-donor__donation">
          {
            data.donorDisplayName
              ? <h4 className="donor__name">{data.donorDisplayName}</h4> : <h4>Anonymous</h4>
          }
          {
            data.donorLocalAmount
              ? (
                <p className="donor__amount">
                  <span>
                    {`£${parseFloat(data.donorLocalAmount).toFixed(2)} `}
                  </span>
                  {
                    data.estimatedTaxReclaim ? (
                      <span>
                        {`+£${parseFloat(data.estimatedTaxReclaim).toFixed(2)}`}
                       Gift Aid
                      </span>
                    ) : null
                  }
                </p>
              ) : null
          }
        </div>
      </div>
      {
        data.message
          ? (
            <div className="donor__statement">
              <p>
                {`"${data.message}"`}
              </p>
            </div>
          ) : null
      }
    </div>
  </div>
);

DonationItem.propTypes = {
  data: PropTypes.object
};

export default DonationItem;
