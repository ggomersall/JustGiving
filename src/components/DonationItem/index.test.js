import React from 'react';
import { shallow } from 'enzyme';
import DonationItem from './index.js';

describe('Testing DonationItem Component', () => {

  let wrapper;

  test('it should render data correctly', () => {
    const data = { 
      imageUrl: 'http://placeholder.com/150',
      donorDisplayName: 'Joe Bloggs',
      donorLocalAmount: 15.00,
      estimatedTaxReclaim: 1.50,
      message: 'lorem ipsum message'
    };
    wrapper = shallow(<DonationItem data={data} />);
  });

  test('Snapshot', () => expect(<DonationItem />).toMatchSnapshot());

});

