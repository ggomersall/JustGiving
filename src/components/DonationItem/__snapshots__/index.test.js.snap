// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing DonationItem Component Snapshot 1`] = `<DonationItem />`;
